src/Tools/GenerateGlobalAnalyzerConfigs/Program.cs

```diff
@@ -371,8 +371,8 @@ static string GetPropertyStringForSettingDefaultPropertyValue(string packageName
                     if (packageName == "Microsoft.CodeAnalysis.NetAnalyzers")
                     {
                         return $@"
-      <!-- Default '{packageVersionPropName}' to 'AnalysisLevel' -->
-      <{packageVersionPropName} Condition=""'$({packageVersionPropName})' == '' and $(AnalysisLevel) != ''"">$(AnalysisLevel)</{packageVersionPropName}>
+      <!-- Default '{packageVersionPropName}' to 'EffectiveAnalysisLevel' with trimmed trailing '.0' -->
+      <{packageVersionPropName} Condition=""'$({packageVersionPropName})' == '' and $(EffectiveAnalysisLevel) != ''"">$([System.Text.RegularExpressions.Regex]::Replace($(EffectiveAnalysisLevel), '(.0)*$', ''))</{packageVersionPropName}>
```

Comment:
If the goal is to remove the trailing 0s couldn't you just do `[System.Int32]::Parse($(EffectiveAnalysisLevel), [System.Globalization.NumberStyles]::Any)`

Reply:
Will that not throw an exception if a user has invalid value for `EffectiveAnalysisLevel`, say some random non-numerical string? I thought this was more safer.

Reply:
I believe a regex that just strips trailing zeros will be safer. I am torn between failing fast for unknown inputs and being as resilient as possible.